### DISCUSSION TITLE
vcpkg in manifest mode uses `VCPKG_TARGET_TRIPLET` only for dependencies but not for the executable itself

### DISCUSSION THREAD

**Discussion by cppdev123** [microsoft/vcpkg#26846] [Q&A] [answered]
I have a project that builds with cmake and have some dependencies much like the example here [manifest example](https://github.com/microsoft/vcpkg/blob/master/docs/examples/manifest-mode-cmake.md).
Now vcpkg be default selects triplet `x64-windows` for everything the executable and dependencies.
I want to build with triplet `x64-windows-static` for everything so I used this camke option when configuring `-DVCPKG_TARGET_TRIPLET=x64-windows-static` and expected vcpkg to use this triplet for the executable and dependencies but it uses this triplet only for dependencies and sticks to `x64-windows` for the executable ! The output always starts with this:

```
Detecting compiler hash for triplet x64-windows... // for the executable
Detecting compiler hash for triplet x64-windows-static... // for dependencies
All requested packages are currently installed. // or install them if not yet installed
```

these are the command lines I used so far without any behavior change :

`cmake .. -DVCPKG_TARGET_TRIPLET=x64-windows-static -DCMAKE_TOOLCHAIN_FILE=E:\programs\vcpkg\scripts\buildsystems\vcpkg.cmake -DVCPKG_TARGET_TRIPLET=x64-windows-static -G "Visual Studio 16 2019" -A x64`

` cmake .. -DVCPKG_TARGET_TRIPLET=x64-windows-static -DCMAKE_TOOLCHAIN_FILE=E:\programs\vcpkg\scripts\buildsystems\vcpkg.cmake -G "Visual Studio 16 2019" -A x64 -DVCPKG_TARGET_TRIPLET=x64-windows-static`

`cmake .. -DVCPKG_TARGET_TRIPLET=x64-windows-static -G "Visual Studio 16 2019" -A x64 -DVCPKG_TARGET_TRIPLET=x64-windows-static -DCMAKE_TOOLCHAIN_FILE=E:\programs\vcpkg\scripts\buildsystems\vcpkg.cmake`

**Comment by Neumann-A** [comment, marked as answer]
set VCPKG_HOST_TRIPLET to the same value as VCPKG_TARGET_TRIPLET

**Reply by cppdev123** [reply]
Thanks this works but it now does not use the installed packages in `vcpkg/installed/x64-windows-static` and instead tries to install new ones inside vcpkg_installed folder in the project directory. I searched and found that this is the default behavior with manifests but I enabled set `VCPKG_DEFAULT_BINARY_CACHE` environment variable and it is already used when I invoke `vcpkg install package` can I use this binary cache for manifests builds ?

**Reply by Neumann-A** [reply]
There is `VCPKG_INSTALLED_DIR` (cmake) to control where stuff gets installed to.
There is `VCPKG_MANIFEST_INSTALL` (cmake) to control if manifest mode is used.

Env variable `VCPKG_DEFAULT_BINARY_CACHE` just controls where vcpkg stores a binary blob of whatever gets installed for reuse if the binary hash is the same so that no rebuild/recompile is necessary for it.  
If vcpkg sees the environment variable `VCPKG_DEFAULT_BINARY_CACHE` it will use it no matter the mode it is in (classic/manifest).

**Reply by cppdev123** [reply]
Another think that the crt linkage in generated visual studio project is still /MD and /MDd no and I have to change it each time

**Reply by Neumann-A** [reply]
that is outside of vcpkg. see CMAKE_MSVC_RUNTIME_LIBRARY
https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html

**Reply by cppdev123** [reply]
yes but how can I determine the runtime library linkage from vcpkg triplet without depending on "static" being in triplet. In other words does vcpkg set a variable I can check to deduce the crt linkage. `VCPKG_CRT_LINKAGE` is empty in the cmake config script

**Reply by bwrsandman** [reply]
Setting `VCPKG_HOST_TRIPLET` does not solve the issue when cross compiling different architectures.
For example, if you tried `arm64-windows`, you'll see that the executable will build x64 objects and try linking with the arm64 dependencies.

**Reply by dg0yt** [reply]
@bwrsandman Maybe it would be better to clearly describe your situation in a separate question instead of implying that this discussion exactly matches your situation.
